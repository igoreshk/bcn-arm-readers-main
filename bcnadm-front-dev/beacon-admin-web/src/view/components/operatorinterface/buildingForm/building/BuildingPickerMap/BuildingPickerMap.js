import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Map, GoogleApiWrapper, Marker } from 'google-maps-react';
import { setBuildingFormField } from 'src/actions/setBuildingFormField';

const MARKER_DEFAULT_LATITUDE = 59.887973;
const MARKER_DEFAULT_LONGITUDE = 30.32818;

export class BuildingPickerMap extends Component {
  constructor(props) {
    super(props);
    this.onMapClick = this.onMapClick.bind(this);
  }

  onMapClick = (ref, map, coords) => {
    const latitude = coords.latLng.lat();
    const longitude = coords.latLng.lng();

    this.props.setBuildingFormField('latitude', latitude);
    this.props.setBuildingFormField('longitude', longitude);
  };

  adjustForZeroCoordinates(latid, long) {
    if (latid && long) {
      return { lat: latid, lng: long };
    }
    return { lat: MARKER_DEFAULT_LATITUDE, lng: MARKER_DEFAULT_LONGITUDE };
  }

  render() {
    const latitude = this.props.buildingFormProps.BuildingForm.values.latitude;
    const longitude = this.props.buildingFormProps.BuildingForm.values.longitude;

    return (
      <Map
        item
        xs={12}
        google={this.props.google}
        onClick={(ref, map, coords) => this.onMapClick(ref, map, coords)}
        zoom={14}
        initialCenter={this.adjustForZeroCoordinates(latitude, longitude)}
        center={this.adjustForZeroCoordinates(latitude, longitude)}
        className="building-picker-map"
      >
        {latitude && longitude && <Marker position={{ lat: latitude, lng: longitude }} />}
      </Map>
    );
  }
}

const mapStateToProps = (state) => {
  return { buildingFormProps: state.form };
};

const mapDispatchToProps = {
  setBuildingFormField
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(
  GoogleApiWrapper({
    apiKey: process.env.GOOGLE_API_KEY
  })(BuildingPickerMap)
);
