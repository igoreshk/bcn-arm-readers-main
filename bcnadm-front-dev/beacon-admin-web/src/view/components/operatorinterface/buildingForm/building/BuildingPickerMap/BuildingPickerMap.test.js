import React from 'react';
import { shallow } from 'enzyme';
import { BuildingPickerMap } from './BuildingPickerMap';

const MARKER_DEFAULT_LATITUDE = 59.887973;
const MARKER_DEFAULT_LONGITUDE = 30.32818;

const props = {
  setBuildingFormField: jest.fn(),
  buildingFormProps: {
    BuildingForm: {
      values: {
        latitude: MARKER_DEFAULT_LATITUDE,
        longitude: MARKER_DEFAULT_LONGITUDE
      }
    }
  }
};

const coords = {
  latLng: {
    lat: jest.fn(),
    lng: jest.fn()
  }
};

const wrapper = shallow(<BuildingPickerMap {...props} />);

describe('BuildingPickerMap', () => {
  const onMapClick = jest.spyOn(wrapper.instance(), 'onMapClick');

  it('Check onMapClick was called', () => {
    wrapper.find('.building-picker-map').simulate('click', null, null, coords);

    expect(onMapClick).toHaveBeenCalled();
  });

  it('Check matching snapshot', () => expect(wrapper).toMatchSnapshot());
});
