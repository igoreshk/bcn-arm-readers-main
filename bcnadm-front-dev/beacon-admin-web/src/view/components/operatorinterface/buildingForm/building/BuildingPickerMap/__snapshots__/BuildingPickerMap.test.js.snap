// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuildingPickerMap Check matching snapshot 1`] = `
<Map
  center={
    Object {
      "lat": 59.887973,
      "lng": 30.32818,
    }
  }
  centerAroundCurrentLocation={false}
  className="building-picker-map"
  containerStyle={Object {}}
  initialCenter={
    Object {
      "lat": 59.887973,
      "lng": 30.32818,
    }
  }
  item={true}
  onClick={[Function]}
  style={Object {}}
  visible={true}
  xs={12}
  zoom={14}
>
  <Marker
    name="Marker"
    position={
      Object {
        "lat": 59.887973,
        "lng": 30.32818,
      }
    }
  />
</Map>
`;
